### ISSUE TITLE
Security Issues

### ISSUE THREAD

**Issue by tunermason** [meshenger-app/meshenger-android#91]
1. GPS location permission. Why is this in there?
2. Inability to add contact without allowing camera access. This needs to be fixed, users need to be able to add contacts without camera access.

**Comment by mwarning** [comment]
Good points!

1. GPS is probably Android permission "ACCESS_FINE_LOCATION", This was added during experiments to access the interfaces MAC address. @adicoder95 do you know more?
2. There is a way to add contacts by copy&paste on the QR code scan screen, but for that you need to allow camera access in the first place. This needs to be fixed.

**Comment by mwarning** [comment]
@tunermason the permission is removed for now, as there seem to be not drawback. Also, the QR code scan site can be used now without camera. Coming for the next release.

**Comment by tunermason** [comment]
Ok thank you, When I compiled the app I commented out the permission without issue. I did face a build issue with CallActivity.kt on like 431. I had to comment out [currentCallSet = true] to get it to build. Overall the code looks good and does not have bloatware or unnecessary dependencies. 

Some features that I think would be useful would be the ability to manually add contacts (key, mac/ip addresses) without needing to use the string. Maybe a form. (expert mode) The ability to change key size, encryption algorithm, hash and codecs.
Renegotiate public keys with a contact(during call or data session. Also more documentation over the network layer (ipv4/6 etc)

How has multi net routing been working?
Meaning client 1 on 10.0.0.0/24 and client 2 on 10.1.0.0/24. Both networks are directly routable. I tried two clients on a OSI layer 3 subnet and they worked so I would imagine that the scenario above would work as well.  

I'd be happy to help where I can.
